cookbooks/continental_extension.prm

```diff
@@ -0,0 +1,256 @@
+#      "Rift Reactivation and migration during multiphase extension"
+
+####  Global parameters
+set Dimension                              = 2
+set Start time                             = 0
+set End time                               = 5e6
+set Use years in output instead of seconds = true
+set Linear solver tolerance                = 1e-7
+set Nonlinear solver scheme                = iterated Stokes
+set Nonlinear solver tolerance             = 1e-4
+set Max nonlinear iterations               = 10
+set Number of cheap Stokes solver steps    = 0
+set CFL number                             = 0.5
+set Output directory                       = output
+set Timing output frequency                = 1
+set Pressure normalization                 = no
+
+#### Parameters describing the model
+
+# Model geometry (400x100 km, 2 km spacing)
+subsection Geometry model
+  set Model name = box
+  subsection Box
+    set X repetitions = 200
+    set Y repetitions = 50
+    set X extent      = 400e3
+    set Y extent      = 100e3
+  end
+end
+
+# Mesh refinement specifications (no mesh refinement)
+subsection Mesh refinement
+  set Initial adaptive refinement        = 0
+  set Initial global refinement          = 0
+  set Time steps between mesh refinement = 0
+end
+
+# Boundary classifications and adiabatic/shear heating specification
+# Velocity: free surface on top, outflow on sides, inflow at base
+# Temperature: fixed at top and bottom, insulating sides
+subsection Model settings
+  set Include adiabatic heating               = false
+  set Include shear heating                   = false
+  set Fixed temperature boundary indicators   = bottom, top
+  set Prescribed velocity boundary indicators = left x: function, right x:function, bottom y:function
+  set Free surface boundary indicators        = top
+end
+
+# Advecting the free surface vertically rather than
+# in the surface normal direction can result in a
+# more stable mesh when the deformation is large
+subsection Free surface
+  set Surface velocity projection = vertical
+end
+
+# Velocity on boundaries characterized by functions
+# The outward velocity (x-direction) on the left and right walls is 0.25 cm/year
+# The vertical velocity at the base is 0.125 cm/year (balances outflow on sides)
+# Velocity components parallel to the base (x-velocity) and side walls (y-velocity)
+# are unconstrained (i.e. 'free'). 
+subsection Boundary velocity model
+  subsection Function
+    set Variable names      = x,y
+    set Function constants  = m=0.0025, year=1
+    set Function expression =  if (x<200e3 , -1*m/year, 1*m/year) ; 0.00125
+  end
+end
+
+
+# Number and name of compositional fields
+# The four compositional fields represent the upper crust, lower crust, mantle
+# and a 'seed' placed in the mantle to help localize deformation.
+subsection Compositional fields
+  set Number of fields = 4
+  set Names of fields = upper, lower, mantle, seed
```

Comment:
I always define my compositional fields from bottom to top, so in the Material model, where the background material (mantle) is by default the first, the order remains the same. Not very important though.